libcaja-private/caja-file.c

```diff
@@ -87,17 +84,12 @@
 #ifdef CAJA_FILE_DEBUG_REF_VALGRIND
 #include <valgrind/valgrind.h>
 #define DEBUG_REF_PRINTF VALGRIND_PRINTF_BACKTRACE
-#else
-#define DEBUG_REF_PRINTF printf
```

Comment:
This is not correct IMO.  `DEBUG_REF_PRINTF` is used 4 times, and it's guarded by `CAJA_FILE_DEBUG_REF` *not* `CAJA_FILE_DEBUG_REF_VALGRIND`.  So although bot `CAJA_FILE_DEBUG_REF` and `CAJA_FILE_DEBUG_REF_VALGRIND` are undef'd on lines 84 and 85 respectively, I would think this serves as debugging switches -- make one or both `#define` and you get the debugging behavior.

In any case, removing this breaks the build with `CAJA_FILE_DEBUG_REF` defined and `CAJA_FILE_DEBUG_REF_VALGRIND` *not* defined.

Reply:
Done, i added the define back.

Reply:
Great :+1: 

Alternatively if the unused macro warning really needs to go, you could actually replace the `#else` with `#elif defined(CAJA_FILE_DEBUG_REF)`, and it should be all good as all use are guarded with `CAJA_FILE_DEBUG_REF`.
I'm happy with the current status, but either way works for me.